extensions/wikia/SitemapXml/tests/SitemapHooksTest.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+class SitemapHooksTest extends WikiaBaseTest {
+	public function setUp() {
+		$this->setupFile = __DIR__ . '/../SitemapXml.setup.php';
+		parent::setUp();
+	}
+
+	/**
+	 * @dataProvider canonicalRedirectProvider
+	 */
+	public function testCanonicalRedirect( $currentRequestUri, $targetUrl, $shouldAllowRedirect ) {
+		$outputMock = $this->getMock( 'OutputPage', [ 'cancelRedirect' ] );
+		$responseMock = $this->getMock( 'WebResponse', [ 'header' ] );
+		$requestMock = $this->getMock( 'WebRequest', [ 'response' ] );
+		$titleMock = $this->getMock( 'Title', [ 'isSpecial', 'getFullURL' ] );
+
+		$requestMock->expects( $this->any() )
+			->method( 'response' )
+			->will( $this->returnValue( $responseMock ) );
+
+		$titleMock->expects( $this->any() )
+			->method( 'isSpecial' )
+			->will( $this->returnValue( true ) );
+
+		$titleMock->expects( $this->any() )
+			->method( 'getFullURL' )
+			->will( $this->returnValue( $targetUrl ) );
+
+		$this->mockStaticMethod( 'WebRequest', 'detectProtocol', parse_url( $currentRequestUri, PHP_URL_SCHEME ) );
+		$this->mockStaticMethod( 'WikiFactoryLoader', 'getCurrentRequestUri', $currentRequestUri );
+
+		$result = SitemapHooks::onTestCanonicalRedirect( $requestMock, $titleMock, $outputMock );
+
+		$this->assertSame( $result, $shouldAllowRedirect );
+	}
+
+	public function canonicalRedirectProvider() {
+		return [
+			[
+				'http://starwars.wikia.com/sitemap-newsitemapxml-index.xml',
+				'https://starwars.fandom.com/sitemap-newsitemapxml-index.xml',
+				false
+			],
+			[
+				'https://starwars.wikia.com/sitemap-newsitemapxml-index.xml',
+				'https://starwars.fandom.com/sitemap-newsitemapxml-index.xml',
+				true
+			],
+			[
+				'http://starwars.fandom.com/sitemap-newsitemapxml-index.xml',
+				'https://starwars.fandom.com/sitemap-newsitemapxml-index.xml',
+				true
+			],
+			[
+				'http://starwars.wikia.com/sitemap-newsitemapxml-index.xml',
+				'https://starwars.wikia.com/sitemap-newsitemapxml-index.xml',
+				true
+			],
+			[
+				'http://wookieepedia.wikia.com/sitemap-newsitemapxml-index.xml',
+				'https://starwars.fandom.com/sitemap-newsitemapxml-index.xml',
+				true
+			],
+			[
+				'https://wookieepedia.wikia.com/sitemap-newsitemapxml-index.xml',
+				'https://starwars.fandom.com/sitemap-newsitemapxml-index.xml',
+				true
+			],
+			[
+				'http://ja.starwars.wikia.com/sitemap-newsitemapxml-index.xml',
+				'https://starwars.fandom.com/sitemap-newsitemapxml-index.xml',
```

Comment:
Are you sure this case is right? `ja.starwars.wikia.com` -> `starwars.fandom.com` without the lang path doesn't really make sense.

Reply:
The point was just to check it won't prevent the redirect when a language subdomain redirects to one without the language path, and I was just using `starwars.fandom.com` for all of them for consistency. This will be the case for some of the exception non-English wikis that may stay at single subdomains, so for example, `de.starwars.wikia.com` is an alias for `jedipedia.wikia.com`. After the switch, unless we decide to move it under `starwars.fandom.com` (these will be reviewed on a case-by-case basis), this will become `jedipedia.fandom.com`. This is a bad example, since the subdomain itself is different, but the test case is just there to make sure a multiple subdomain alias does not match if the language code is not there on the redirect target.

Reply:
Makes sense, thanks for the explanation!